### PR TITLE
Add submitReport API

### PR DESCRIPTION
**Pull request by GiovanniGobbs**
- Add submitReport API
- Create API for retrieving the website's icon (see APIService.java)

### CODE REVIEW

File: app/src/main/java/com/uncookthebook/app/network/APIService.java [hunk 1]
```diff
@@ -16,8 +19,14 @@
...
+    Call<String> addUser(@Body TokenizedObject<User> request);
 
     @POST("/utb/get_article")
     Call<GetArticleResponse> getArticle(@Body TokenizedObject<GetArticleRequest> request);
+
+    @POST("/utb/submit_report")
+    Call<String> submitReport(@Body TokenizedObject<Report> request);
+
+    @GET("{url}")
+    Call<String> getWebsiteIcon(@Path("url") String url);
```
Comment (matteofiordarancio) [inline app/src/main/java/com/uncookthebook/app/network/APIService.java:31]: I use directly the okHttpClient

File: app/src/main/java/com/uncookthebook/app/models/User.java [hunk 2]
```diff
@@ -11,7 +11,7 @@
  */
 @Getter
 @ToString
-public final class User implements Model {
+public class User implements Model {
```
Comment (matteofiordarancio) [inline app/src/main/java/com/uncookthebook/app/models/User.java:14]: why did you remove the final? I fail to see where you need to reuse it

**Review by matteofiordarancio — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/uncookthebook/app/models/User.java (hunk 2 above)
Comment (GiovanniGobbs) [reply app/src/main/java/com/uncookthebook/app/models/User.java:14]: Because I didn't put it in the others and so I removed it. I'll add it in every Model class

File: app/src/main/java/com/uncookthebook/app/network/APIService.java (hunk 1 above)
Comment (GiovanniGobbs) [reply app/src/main/java/com/uncookthebook/app/network/APIService.java:31]: Why? It's easier to use this

**Review by matteofiordarancio — APPROVED** [review]
